packages/toolpad-components/src/Map.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import { SxProps } from '@mui/material';
+import Map from 'react-map-gl';
+import { createComponent } from '@mui/toolpad-core';
+
+interface MapComponentProps {
+  longitude: number;
+  latitude: number;
+  zoom: number;
+  sx: SxProps;
+  style: string;
+}
+
+const MapComponent = ({ latitude, longitude, zoom, sx, style }: MapComponentProps) => {
+  return (
+    <Map
+      key={`${latitude}-${longitude}-${zoom}`}
+      initialViewState={{ latitude, longitude, zoom }}
+      style={sx as React.CSSProperties}
+      mapStyle={style}
+      mapboxAccessToken={
+        'pk.eyJ1IjoibWFib25nIiwiYSI6ImNrMm9qN2tiYTEwc3ozZG41emx6bW9uZnQifQ.PhojWq3UwsAlPB7LBvJiTw'
```

Comment:
would it make sense to place it in some config file? 🤔

Reply:
Yes, for now I didn't add it as an environment variable for the ease of being able to view it in the render.com preview; I wasn't sure how we would add environment variables to code in `toolpad-components` folder on render.com

Reply:
The easiest way for now would probably be by using a build time variable.
- add a key [here](https://github.com/mui/mui-toolpad/blob/299a8ccf1af5e09f44c715f41579057470b7d1a5/packages/toolpad-app/src/config.ts#L25)
- pass the var [here](https://github.com/mui/mui-toolpad/blob/299a8ccf1af5e09f44c715f41579057470b7d1a5/packages/toolpad-app/next.config.mjs#L25)
- Use it in your code as `process.env.MAPBOX_ACCESS_TOKEN` (exactly like that, so that webpack can find/replace it)

[Here's](https://github.com/mui/mui-toolpad/blob/299a8ccf1af5e09f44c715f41579057470b7d1a5/packages/toolpad-app/src/config.ts#L4) some more info about the different ways configuration can be passed to Toolpad. (For this case, only 1. will work. 3. doesn't apply as these are browserside components. If we want 2., then we're going to need to add a context or something for it in `toolpad-core`)